### PR TITLE
feat: add html editor

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio/issues/2359 https://github.com/webstudio-is/webstudio/issues/2388

Here added codemirror based html editor for html embed component. It supports basic autocomplete and highlights embedded css and js.

Splitted basic codemirror setup into CodeEditor component everything else can be customized around it.

Screenshot 2023-12-27 at 15 30 57
Screenshot 2023-12-27 at 15 30 22

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

**Comment by kof** [comment]
This other issue is also about using mono font variant for cli code, so probably don't close it yet https://github.com/webstudio-is/webstudio/issues/2388

**Comment by kof** [comment]
- [ ] regression: initial size needs to be much biger as it used to be

File: apps/builder/app/builder/features/settings-panel/controls/code.tsx [hunk 1]
```diff
@@ -11,31 +18,46 @@ export const CodeControl = ({
...
+    ...meta,
+    control: "text" as const,
+  };
+  const localValue = useLocalValue(prop?.value ?? "", (value) => {
+    // sanitize html before saving
+    // this is basically what browser does when innerHTML is set
+    // but isolated within temporary element
+    // so the result is correct markup
+    const div = document.createElement("div");
+    div.innerHTML = value;
```
Comment (kof) [inline apps/builder/app/builder/features/settings-panel/controls/code.tsx:31]: is there risk of executing scripts etc? I don't remember if it does ...

Comment (TrySound) [reply apps/builder/app/builder/features/settings-panel/controls/code.tsx:31]: innerHTML doesn't execute scripts

**Review by kof — APPROVED** [review]
please don't forget to set min height, very important, looks awkward otherwise

**Pull request merged**